research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs

```diff
@@ -451,3 +468,41 @@ fn get_partition_vertex_ids(
     let partition_id = graph_partition_manager.get_partition_id(id as VertexId) as PartitionId;
     vec![(partition_id, vec![id as VertexId])]
 }
+
+/// Get vertex ids by giving label_ids and corresponding primary key values
+fn get_vertex_ids_by_primary_keys(
+    labels: &Vec<LabelId>,
+    graph_partition_manager: Arc<dyn GraphPartitionManager>,
+    primary_key_list: Vec<String>,
```

Comment:
Why primary_key is a string type? property has been transformed as id no?

Reply:
Different primary key types (e.g., `int64`, `string`) are handled by a same FFI function, and a string argument is expected. For int types, we first cast it to string by `to_string()`, e.g., `1` -> `"1"`.